packages/ember-data/lib/system/store.js

```diff
@@ -1446,11 +1447,7 @@ Store = Service.extend({
   },
 
   modelFactoryFor: function(key) {
-    if (this.container.has('model:' + key)) {
-      return this.container.lookupFactory('model:' + key);
-    } else {
-      return null;
-    }
+    return this.container.lookupFactory('model:' + key);
```

Comment:
@igorT getting rid of the `has` check did not break any tests. You added this for the polymorphic mixins so I wanted to get your eyes on this change.